integration-tests/src/test/resources/features/AddressUnhappyPath.feature

```diff
@@ -41,3 +41,9 @@ Feature: Address API unhappy path test
     Given a request is made to the postcode-lookup endpoint without a postcode in the body and with session id in the header
     Then the endpoint should return a 400 HTTP status code
     And the response body contains no postcode error
+
+  @invalid_date
+  Scenario: Date year in the request body is the invalid value 0000
```

Comment:
This is just one part of the regex, right? Shall we also add a test for the other parts too?